overlord/snapstate/snapstate.go

```diff
@@ -734,7 +734,7 @@ func doUpdate(st *state.State, names []string, updates []*snap.Info, params func
 	}
 
 	// first snapd, core, bases, then rest
-	sort.Sort(byKind(updates))
+	sort.Sort(byKindAndName(updates))
```

Comment:
which tests do exactly need this change?

Reply:
This might be relevant in this context: https://github.com/snapcore/snapd/pull/5708

Reply:
yea,  I discussed with @bboozzoo to try to avoid the need for the change because ideally we want to switch to use ByType here, otherwise we can first sort by name and then do a stable sort with ByType but feels overkill as discussed